consensus/consensus.go

```diff
@@ -170,14 +173,14 @@ func VerifyBlock(bc *blockchain.BlockChain,
 	}
 
 	// Check that block number is valid.
-	ix := b.BlockNumber - 1
-	if int(ix) > len(bc.Blocks)-1 || ix < 0 {
+	lastBlockIdx := b.BlockNumber - 1
+	if int(lastBlockIdx) != len(bc.Blocks)-1 {
```

Comment:
I would change the comment on this function to say something about the fact that we are making sure that this block can be added to the _end_ of the chain.